### DISCUSSION TITLE
Z-Stack_3.x.0 coordinator 20230901/20230902 feedback

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/Z-Stack-firmware#474] [Polls]
**Update 20230923: based on the feedback we can conclude that the 20230901/20230902 firmwares don't work well. Let's take a step back and start from 20221226, I've compiled two new test firmwares: https://github.com/Koenkk/Z-Stack-firmware/discussions/478**

Since some users are having issues with the [20230507](https://github.com/Koenkk/Z-Stack-firmware/issues/439) firmware which they didn't have with 20221226, I've compiled 2 new test firmwares.

Download: [20230901_20230902.zip](https://github.com/Koenkk/Z-Stack-firmware/files/12503459/20230901_20230902.zip)

When providing feedback:
- **First confirm that after flashing the firmware, no missing routers are found with the [coordinator check](https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-request-coordinator-check)**
- Fill in the poll
- Copy and comment the template below:

```
- Amount of devices in network:
- Bridge software (Zigbee2MQTT/ZHA):
- 20221226 worked fine: yes/no
- 20230901 feedback:
- 20230902 feedback:

Zigbee herdsman debug log in case of complete crash (Zigbee2MQTT only, link to file): 
```

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable the herdsman debug logging. Note that this is only logged to STDOUT and not to log files.

**Comment by ahd71** [comment]
Was about to test but the missing router check shows all, or almost all, my routers as missing.  Haven't flashed yet (but is on the 20230716 test release), can uodate and check again.

But I guess its time for me to start all over and build a new network since the recent outage started getting me problems since some few weeks. The network is slower than usually and hard to join new devices so matches the listed symtoms.

Curious about the diff between the two september releases. Intentional not exposed to not bias the survey response?

**Reply by Koenkk** [reply]
> But I guess its time for me to start all over and build a new network since the recent outage started getting me problems since some few weeks. The network is slower than usually and hard to join new devices so matches the listed symtoms.

Repairing the missing routers should yield the same effect as repairing everything. Z2M 1.33.0 contains a potential fix to prevent routers from missing again. So I recommend to just repair the missing routers instead.

> Curious about the diff between the two september releases. Intentional not exposed to not bias the survey response?

Some tables are sized differently, for the rest no difference. Let's see which one performs the best 😄

**Reply by snippem** [reply]
@Koenkk i am getting stuck at the coordinator_check (Did not even now this was possible).
What is the best way to send this command ? i have Mqtt explorer but what to enter as a message exactly? or is there another way to do this?  I have read https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html
but still get stuck haha. Sorrie for getting a bit of topic.

**Reply by adekloet** [reply]
Are you using Home assistant? If yes, go to the MQTT integration and press configure, on the next screen you can publish a packet in the MQTT topic and below you can listen to the topic for the results.

below is an example, in this case I am requesting a networkmap, but just replace the topic with the one for the check ( zigbee2mqtt/bridge/request/coordinator_check).

![afbeelding](https://github.com/Koenkk/Z-Stack-firmware/assets/60182504/de7ef288-7914-4ed6-b2c4-d0bd94da2ce3)

**Reply by snippem** [reply]
Thanks :-) this works published a package with homeassistant mqtt and checked with mqtt explorer and fixed the missing routers.

**Comment by sjorge** [comment]
02 is having so many issues for me right off the get go, my legrand enegry meter just stops responding. No missing routers before and after flashing. Will try 01 when i get back home.

No NWK_TABLE_FULLs tough, but 2 sensors also seem to be no longer responding, but that is a bit harder to tell when not at home.

Was not on 26 before so can't attest to that one, was originally not going to comment because if this. But 02 has heen the worst I've had in a while.

Edit:

01 so far behaving normally, if it's still normal tomorrow I'll go back to 02

**Reply by sjorge** [reply]
Some more back and forth between 02 and 01…

02 just drops device, well just stop seeing any traffic to them, even when sniffing with wireshark. 01 seems fine, not ran it for weeks yet so will do that now.

**Reply by Koenkk** [reply]
> well just stop seeing any traffic to them

From device to coord or coord to device?

**Reply by sjorge** [reply]
Coord -> device, i still saw device -> coord messages, with wireshark, nothing on z2m side. (Was watching a herdsman debug, don’t think they are making it up that far if the coord even receives them)

my 2 device test network, is ofcourse fine in 02 😑 so it’s a bit painful to mess with my main one.

let’s see how 01 does, far it seems fine on ~ 2 day ish uptime.

**Reply by Koenkk** [reply]
is it just a single device? if yes, can you try repairing that device?

**Reply by sjorge** [reply]
It's not always the same one, It's often the ones that talk the most though.
I already tried repairing one of the stuck ones,but while the 02 has the stuck behavior I can't repair it.

Restarting z2m doesn't help in that case, the router check still says the coord knows about it if I run it at the time it's stuck.

**Reply by sjorge** [reply]
@Koenkk so far 01 has been pretty stable, no NWK_TABLE_FULL or missing devices.

**Comment by BradleyFord** [comment]
I have been running 20230902 since the day of this release:

It worked for a while, but then I started loosing devices again. But interestingly it dropped down to 49 devices and just stayed at that level for about 12 hours until I did the usual reset and it jumped back up to the normal 70 range.

I found some devices would come back online if I did a reinterview...

- Amount of devices in network:
-- registered: 106
-- Normally online: 70 (Most are lights, of which some I am still running as dumb lights until I can get this fixed)
- Bridge software (Zigbee2MQTT/ZHA): Zigbee2MQTT
- 20221226 worked fine: yes
- 20230901 feedback: ??
- 20230902 feedback: No


Note: 1 hour after the reset whilst writing this comment, it again started dropping devices. It got to 58 before I reset it again

Note2:
Thinking about this more, the nodes that survived are withing range of the coordinator...and I think 50 is the limit for direct connections. So it seems the ones that I lost all needed to be routed to get to the coordinator

**Reply by Koenkk** [reply]
Please also try `20230901`

**Comment by Koterak** [comment]
Using 20230902 from first day and no problems at all. I have 148 devices.
All respond quick, and no missing routers

wt., 5 wrz 2023 o 19:36 Koen Kanters ***@***.***> napisał(a):

**Comment by cloudbr34k84** [comment]
man im scared to try this firmware as 20230716 has been really good for me

**Reply by Koenkk** [reply]
Then try 20230901 which is nearly the same. Basically 20221226, 20230716 and 20230901 are equal except the SDK version.

**Comment by adekloet** [comment]
Just a short update from me, I have been running 20230716 for two weeks now and have not seen any major delays or cut outs from the coordinator. On the [20230507](https://github.com/Koenkk/Z-Stack-firmware/commit/c80b77f8083f729efdf3fbac243c2d88b935ebfb) version I had two complete outages, clearly seen on the MQTT mem usage graph:

![image](https://github.com/Koenkk/Z-Stack-firmware/assets/60182504/079f3282-4360-487a-8f62-b891a5f50fc9)

I had no significant log files when that happend. The network just became onresponsive with the Network table full Error (as described iin the 20230507 thread.

I now also have a flashed zigbee Sniffer stick available and have wireshark ready to capture data from the network to see what the messaging is when a failure occurs (I have automations in HA to notify me on high Mem for Z2MQTT usage etc.)

Sofar the network is stable with 101 devices. I have last seen configured and have active reporting when I don't see devices reporting for more then 24 hours. In the two weeks I have had to re-pair two end sensor devices sonoff contact sensors. These could also have been running low on batt, so I am monitoring for now . No routers have disconnected.

Zigbee2MQTT version
    [1.33.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.33.0) commit: [unknown](https://github.com/Koenkk/zigbee2mqtt/commit/unknown)

Coordinator type
    zStack3x0

Coordinator revision
    20230716

Frontend version
    0.6.135

Total 101

**By device type**
Router: 67
End devices: 34

**By power source**
Mains (single phase): 67
Battery: 32
undefined: 2

**By vendor**
IKEA of Sweden: 22
OSRAM: 11
LEDVANCE: 8
LUMI: 7
_TZ3000_kdi2o9m6: 7
_TZ1800_fcdjzz3s: 5
eWeLink: 5
TexasInstruments: 4
_TZ3000_g5xawfcq: 3
_TZ3000_1obwwnmq: 3
_TZE200_uebojraa: 2
_TZ3000_i8l0nqdu: 2
_TZ3000_mraovvmm: 2
_TZ3000_gvn91tmx: 2
_TZE200_zah67ekd: 2
TUYATEC-yg5dcbfu: 1
TUYATEC-3tipnsrx: 1
_TZ3000_ukuvyhaa: 1
_TZ3000_xabckq1v: 1
_TYZB01_qeqvmvti: 1
_TZ3000_8nkb7mof: 1
3A Smart Home DE: 1
SONOFF: 1
XIAOMI: 1
undefined: 1
_TZ3000_4fjiwweb: 1
_TZ3000_zloso4jk: 1
_TZ3000_fllyghyj: 1
_TZ3000_upgcbody: 1
_TZ3000_u5u4cakc: 1
_TZ3000_amdymr7l: 1

**By model**

TS011F: 17
Plug 01: 12
TY0202: 5
lumi.weather: 5
TS0601: 4
ti.router: 4
DS01: 4
TS0121: 4
TRADFRI motion sensor: 3
TRADFRI control outlet: 3
CLA60 RGBW Z3: 3
Plug Z3: 3
TRADFRI bulb GU10 WS 400lm: 3
TRADFRI bulb E14 WS opal 400lm: 2
TS0503B: 2
TS004F: 2
TRADFRI bulb GU10 W 400lm: 2
TRADFRI on/off switch: 2
TRADFRI bulb GU10 WW 400lm: 2
TRADFRI bulb E27 WW clear 250lm: 2
Gardenpole Mini RGBW Z3: 2
TRADFRI bulb E27 WS opal 980lm: 1
RH3052: 1
lumi.router: 1
RH3001: 1
TS0504B: 1
TRADFRI Driver 10W: 1
TH01: 1
TS0011: 1
LXN56-DS27LX1.3: 1
DONGLE-E_R: 1
lumi.sen_ill.mgl01: 1
lumi.sensor_motion.aq2: 1
TRADFRI remote control: 1
TS0201: 1
TS0207: 1

**Reply by adekloet** [reply]
Onging test Update 14-Sept-2023 -> 20230716 ( on a https://slae.sh/projects/cc2652/ ) still runs without any problem (No network table full or other zigbee network crashes). The one door sensor repaired and shorted the reporting period. No devices dropped yet. 20230716 = 20230901 (only older SDK version) as reported by Koen. 

I am curious if anybody had NETWORK TABLE FULL errors or a completely inresponsive Zigbee network which either needed a Z2MQTT restart (I run the add-on on HASSOS - RPI4) or had to reseat the zigbee stick on 20230901?

**Reply by deviantintegral** [reply]
> I am curious if anybody had NETWORK TABLE FULL errors

I'm on a different stick (zzh!), but I looked through my logs and I've not had any NWK_TABLE_FULL errors in the past 3 months, _except_ for devices that are actually unavailable.

**Comment by alexruffell** [comment]
Sorry I can't provide logs and/or do time consuming troubleshooting, but I did test the 3 versions of the firmware on my Sonoff Zigbee 3.0 USB P dongle and the best firmware is the 20221226 as I get the least "unregistered endpoint" errors and non responsive devices. 20230901 worked well for a few days then I started noticing failed devices so I moved to 20230902 and it was a total disaster. Half my network was dead.

I have 86 zigbee devices with 50 routers and 36 battery operated devices. I use ZHA and am on the latest version of HA at all times.

I have tested all versions since 20221226 and none have worked well for me. :(

EDIT: I downgraded back to 20221226 and everything came back and is working well.

EDIT2:

- Amount of devices in network: 86 (50 routers (mostly outlets/switches); 36 battery operated devices)
- Bridge software (Zigbee2MQTT/ZHA): ZHA
- 20221226 worked fine: yes
- 20230901 feedback: worked for a few days (or so it seemed) and then back to lots of unregistered endpoints and failures
- 20230902 feedback: hell within a short period of time. Nothing seemed to work and half the network was offline.

EDIT3:

Just noticed this graph where light blue is ESPHome so not Zigbee. The other two are in a well covered spot and are zigbee. The few days before they started working fine I was on the new builds so they were very unreliable. Once I downgraded... perfect again.
<img width="392" alt="image" src="https://github.com/Koenkk/Z-Stack-firmware/assets/4006944/f4aa97a2-148e-45b8-83fa-29aa11af5351">

**Comment by Venom84** [comment]
Hi,
I have also been having lots of issues with FW 20230507 in that my Zigbee2MQTT seems to crash and not be able to connect to the coordinator and the only way to resolve this is to disconnect the coordinator USB and reinsert it again.

I have just flashed 20230901 and will monitor and report back.

**Comment by snippem** [comment]
Have the 20230902 now running for a week.
My experience is:
No missing routers.
2 sensors are not responding.

Fixing the missing routers did solve a lot off problems I had. The missing routers where fixed right after flashing 20230902. But for know me and my family's experience is positive with the 20230902 firmware.  I will fix the two sensors and repair the missing routers if they apeare after flashing 20230901.
Will give it a week and see how that goes.

My network:
Total 107 devices , 34 routers and 73 battery devices.

**Comment by popy2k14** [comment]
Tested also all firmware versions since 20221226.

- Amount of devices in network: 129, see details at the bottom of the post
- Bridge software (Zigbee2MQTT/ZHA): Zigbee2MQTT 1.33.0
- 20221226 worked fine: yes a long time until i flashed 20230507 (but Z2M failed yesterday after 6 days runtime, sadly no debug log)
- 20230901 feedback: flashed today and will keep an eye on it...
- 20230902 feedback:

What i can say about the other versions:
 - 20230507 feedback: not stable, often slows down
 - 20230716 feedback: stable for me, has not sloed down for me

I hope we rule this out together, to get an stable base again.

```
Gesamt 129
Nach Typ
Endgerät: 86
Router: 43
Nach Stromquelle
Batterie: 86
Netz (einphasig): 43
Nach Hersteller
Philips: 42
Signify Netherlands B.V.: 36
LUMI: 31
IKEA of Sweden: 11
Heiman: 4
HEIMAN: 3
Develco: 1
_TZE200_s8gkrkxk: 1
Nach Modell
lumi.sensor_magnet.aq2: 25
RWL021: 24
LWB010: 12
SML001: 9
LTW013: 6
LTW010: 6
LST002: 5
FYRTUR block-out roller blind: 4
SMOK_YDLV10: 4
LTW012: 4
KADRILJ roller blind: 4
lumi.sensor_motion.aq2: 3
lumi.sensor_switch: 3
TRADFRI open/close remote: 3
LCT015: 3
LCL001: 2
SML003: 2
LCT007: 2
SmokeSensor-N-3.0: 2
SML002: 1
LWG004: 1
ZHEMI101: 1
SmokeSensor-EF-3.0: 1
LTA001: 1
TS0601: 1
```

**Comment by JimKlapwijk** [comment]
Hi. I use 20230507. But ever since I did that I have 2 servers which disconnects the USB.

```
Sep  8 12:38:46 kvm-01 kernel: [861836.222249] usb 1-9: new full-speed USB device number 5 using xhci_hcd
Sep  8 12:38:47 kvm-01 kernel: [861836.371562] usb 1-9: New USB device found, idVendor=1a86, idProduct=7523, bcdDevice= 2.64
Sep  8 12:38:47 kvm-01 kernel: [861836.371567] usb 1-9: New USB device strings: Mfr=0, Product=2, SerialNumber=0
Sep  8 12:38:47 kvm-01 kernel: [861836.371569] usb 1-9: Product: USB Serial
Sep  8 12:38:47 kvm-01 kernel: [861836.372844] ch341 1-9:1.0: ch341-uart converter detected
Sep  8 12:38:47 kvm-01 kernel: [861836.373296] usb 1-9: ch341-uart converter now attached to ttyUSB1
Sep  8 12:38:47 kvm-01 kernel: [861836.719288] ch341-uart ttyUSB1: ch341-uart converter now disconnected from ttyUSB1
Sep  8 12:38:47 kvm-01 kernel: [861836.719305] ch341 1-9:1.0: device disconnected
Sep  8 12:38:47 kvm-01 kernel: [861836.958462] usb 1-9: reset full-speed USB device number 5 using xhci_hcd 
```

I use a CC2652R1 (from ZZH) with Debian 10:
`Linux kvm-02 5.15.116-1-pve #1 SMP PVE 5.15.116-1 (2023-08-29T13:46Z) x86_64 GNU/Linux`

Would upgrading to any of the above fix this issue? 
Thank you! 

EDIT: I thought f*** it, I'll give it a go, what could possibly go wrong, so I upgraded to 20230902 and this fixed the above problem for me on the 2 machines :).

**Comment by Wickedy123** [comment]
I've been experiencing a couple of months of stability on 20230716, until trying to charge a supposedly low power bluetooth earpiece, which tripped out all the USB ports (entirely my fault - should have found a real charger!)

But this got me thinking - the stability I've encountered roughly coincides with me not using any of the spare the USB ports on the Pi for anything. So I have highly anecdotal (and circumstantial) evidence that use of the other USB ports on the pi might affect the performance of my Sonoff Dongle. 

To anyone who HAS been experiencing issues with 20230716:
- Are your USB ports all empty, aside from the Coordinator and maybe the SDD?
- Do you have the SDD connected to a USB 3.0 port (blue socket) and the Coordinator to a 'standard' USB port? (this is my setup)

I am more than willing to get shot down on this theory by anyone who knows more on this subject (which is probably most of you :-) ) As always, I'm just trying to be helpful.

Ps. apologies for not trying out the 0902 versions yet.

**Comment by catduckgnaf** [comment]
Is there a version for the CC2652P7 ?

**Comment by dlasher** [comment]
- Amount of devices in network: 75 (End devices: 46 Router: 29)
- Bridge software (Zigbee2MQTT/ZHA): [1.32.2]  commit: [1ec1e57]
- Adapter - HamGeek ESP32/CC2652P POE
- 20221226 worked fine:  no - required power cycle
- 20230507 feedback: random crashes, but reboots itself.
- 20230901 feedback: still crashing
- 20230902 feedback: still crashing

```
error 2023-09-11 13:28:57: Adapter disconnected, stopping
error 2023-09-11 13:47:58: Adapter disconnected, stopping
error 2023-09-11 14:18:59: Adapter disconnected, stopping
error 2023-09-11 19:59:08: Adapter disconnected, stopping
error 2023-09-12 01:14:17: Adapter disconnected, stopping
error 2023-09-12 08:06:31: Adapter disconnected, stopping
error 2023-09-12 09:32:33: Adapter disconnected, stopping
```

Zigbee herdsman debug log in case of complete crash (Zigbee2MQTT only, link to file): (debug is running now, hope to get something useful from the next one)

```
 zigbee-herdsman:adapter:zStack:znp:AREQ <-- AF - incomingMsg - {"groupid":0,"clusterid":1794,"srcaddr":41587,"srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"linkquality":98,"securityuse":0,"timestamp":11400555,"transseqnumber":0,"len":9,"data":{"type":"Buffer","data":[8,106,10,0,4,42,137,1,0]}} +384ms
  zigbee-herdsman:controller:log Received 'zcl' data '{"frame":{"Header":{"frameControl":{"frameType":0,"manufacturerSpecific":false,"direction":1,"disableDefaultResponse":false,"reservedBits":0},"transactionSequenceNumber":106,"manufacturerCode":null,"commandIdentifier":10},"Payload":[{"attrId":1024,"dataType":42,"attrData":393}],"Command":{"ID":10,"name":"report","parameters":[{"name":"attrId","type":33},{"name":"dataType","type":32},{"name":"attrData","type":1000}]}},"address":41587,"endpoint":1,"linkquality":98,"groupID":0,"wasBroadcast":false,"destinationEndpoint":1}' +450ms
  zigbee-herdsman:controller:endpoint DefaultResponse 0xbc33acfffe3522ba/1 1794(10, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":1,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) +455ms
  zigbee-herdsman:adapter:zStack:adapter sendZclFrameToEndpointInternal 0xbc33acfffe3522ba:41587/1 (0,0,1) +455ms
  zigbee-herdsman:adapter:zStack:znp:SREQ --> AF - dataRequest - {"dstaddr":41587,"destendpoint":1,"srcendpoint":1,"clusterid":1794,"transid":226,"options":0,"radius":30,"len":5,"data":{"type":"Buffer","data":[24,106,11,10,0]}} +456ms
  zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,15,36,1,115,162,1,1,2,7,226,0,30,5,24,106,11,10,0,116] +456ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [] +12ms
Zigbee2MQTT:info  2023-09-12 10:12:29: MQTT publish: topic 'zigbee2mqtt_solo/Kitchen.Under.Cabinet', payload '{"brightness":254,"device":{"applicationVersion":18,"friendlyName":"Kitchen.Under.Cabinet","hardwareVersion":2,"ieeeAddr":"0xbc33acfffe3522ba","manufacturerID":4388,"manufacturerName":"Jasco Products","model":"43082","networkAddress":41587,"powerSource":"Mains (single phase)","stackVersion":101,"type":"Router","zclVersion":3},"energy":8.12,"last_seen":"2023-09-12T10:12:29-07:00","linkquality":98,"power":39.3,"state":"ON"}'
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [254,1,100,1,0,100] +18ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,1,100,1,0,100] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser --> parsed 1 - 3 - 4 - 1 - [0] - 100 +0ms
  zigbee-herdsman:adapter:zStack:znp:SRSP <-- AF - dataRequest - {"status":0} +456ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [254,3,68,128,0,1,226,36] +46ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,3,68,128,0,1,226,36] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser --> parsed 3 - 2 - 4 - 128 - [0,1,226] - 36 +0ms
  zigbee-herdsman:adapter:zStack:znp:AREQ <-- AF - dataConfirm - {"status":0,"endpoint":1,"transid":226} +76ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [254,11,69,200,115,162,186,34,53,254,255,172,51,188,3,219] +2s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,11,69,200,115,162,186,34,53,254,255,172,51,188,3,219] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser --> parsed 11 - 2 - 5 - 200 - [115,162,186,34,53,254,255,172,51,188,3] - 219 +0ms
  zigbee-herdsman:adapter:zStack:znp:AREQ <-- ZDO - concentratorIndCb - {"srcaddr":41587,"extaddr":"0xbc33acfffe3522ba","pktCost":3} +2s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [] +1ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [254,11,69,200,115,162,186,34,53,254,255,172,51,188,2,218] +1s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,11,69,200,115,162,186,34,53,254,255,172,51,188,2,218] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser --> parsed 11 - 2 - 5 - 200 - [115,162,186,34,53,254,255,172,51,188,2] - 218 +0ms
  zigbee-herdsman:adapter:zStack:znp:AREQ <-- ZDO - concentratorIndCb - {"srcaddr":41587,"extaddr":"0xbc33acfffe3522ba","pktCost":2} +1s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [] +0ms
  zigbee-herdsman:adapter:zStack:znp:log Port closed +13m
  zigbee-herdsman:controller:log Adapter disconnected' +5s
Zigbee2MQTT:error 2023-09-12 10:12:34: Adapter disconnected, stopping
Zigbee2MQTT:info  2023-09-12 10:12:34: MQTT publish: topic 'zigbee2mqtt_solo/bridge/state', payload '{"state":"offline"}'
Zigbee2MQTT:info  2023-09-12 10:12:34: Disconnecting from MQTT server
Zigbee2MQTT:info  2023-09-12 10:12:34: Stopping zigbee-herdsman...
```

Not a lot there to work with.

**Reply by dlasher** [reply]
I have a Sonoff 3.x stick in a drawer, pulled it out, flashed it to 2023507 - going to run that for a while, try to eliminate hardware as the sole cause.

**Reply by Koenkk** [reply]
Your firmware doesn't look to be crashing, seems the device is disconnected, this usually occurs because of power supply issues or other processes accessing the port.

**Reply by dlasher** [reply]
The original adapter (HamGeek ESP32/CC2652P POE) is 10ft from the POE switch, connected to a POE/POE+/POE++ port, so in theory it shouldn't be power. And nothing else on my network talks to it on that VLAN, except the Zigbee2MQTT LXC container. 

I do see this discussion : https://github.com/Koenkk/zigbee2mqtt/discussions/17360
```
It simply stops providing output and hangs Z2M. Quite noticeable in my case because part of home automation stops working and can hear audible alarm (wife screaming... )
```

Which makes me wonder whether it's just "too cheap" and I need to find a higher quality version. Any personal recommendations for a POE/Ethernet 3.x coordinator you like?

**Reply by cloudbr34k84** [reply]
> The original adapter (HamGeek ESP32/CC2652P POE) is 10ft from the POE switch, connected to a POE/POE+/POE++ port, so in theory it shouldn't be power. And nothing else on my network talks to it on that VLAN, except the Zigbee2MQTT LXC container. 
> 
> I do see this discussion : https://github.com/Koenkk/zigbee2mqtt/discussions/17360
> ```
> It simply stops providing output and hangs Z2M. Quite noticeable in my case because part of home automation stops working and can hear audible alarm (wife screaming... )
> ```
> 
> Which makes me wonder whether it's just "too cheap" and I need to find a higher quality version. Any personal recommendations for a POE/Ethernet 3.x coordinator you like?
> 
> 

I have the hamgeek and have never had this issue. Occasionally I get the 6000 error unsure why and just restart. Mine is not powered by Poe. Maybe try powering it but the USB cable and just plug it into a non Poe port for data ?

**Comment by deviantintegral** [comment]
I installed 20230901 on my zzh! last week.

- Amount of devices in network: 109
- Bridge software (Zigbee2MQTT/ZHA): zigbee2mqtt
- 20221226 worked fine: no, but it was also long enough ago other issues could have been at play. Would you like people to re-test with this firmware too?
- 20230901 feedback: So far no full crashes. However, I've had greater overall instability with lags and missing messages. One Ikea bulb consistently goes unavailable for an hour each day. I noticed an Enbrighten switch not correctly responding to an on command. I'm surprised this is even possible, but I've also had issues with a Philips dimmer switch not working over direct binding. After 10 seconds or so of pressing buttons it starts working agian.
- 20230902 feedback: More unstable than 20230901. The enbrighten switch would go unavailable multiple times per day. I didn't experience full crashes, but I only tested for 24 hours before reverting back to 20230716.

I appreciate all the work here.

**Reply by deviantintegral** [reply]
I edited to add my notes on 20230902.

Of note is that I've never once had coordinator check return any missing routers.

**Reply by zen2** [reply]
About 20221226 version:

My zigbee network is made up of 103 devices ( 62 end devices, 41 routers ).
I've been using this version for as long as it's existed, so I've been using it for 8.5 months now without any kind of problems.
I've followed all the problems linked to new firmwares and I've decided to stick with this one for day-to-day stability.

**Comment by smulle48** [comment]
- Amount of devices in network: 31
- Bridge software (Zigbee2MQTT/ZHA): Zigbee2MQTT
- 20221226 worked fine: yes
- 20230901 feedback: worked most of the time for about 5 days. Then ghost devices started appearing and the network shut of randomly and don't restart by itself
- 20230902 feedback: not tested

**Comment by cloudbr34k84** [comment]
Amount of devices in network: 124
Bridge software (Zigbee2MQTT/ZHA): Zigbee2MQTT
20230716: i felt this was really good untill the last few days where i had complete 6000 errors few times a day
20221226 worked fine: yes

### 20230901: 
update about 6 hrs ago from writing this. Had my first major issue - Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms). Completly stop Z2M add on. The restarts my zigstar POE adapter. See how we go overnight. ALso seeing a lot of errros on devices that have been fine on other firmware. See log for errors
[log.txt](https://github.com/Koenkk/Z-Stack-firmware/files/12672353/log.txt)

- **Update on 20230901: All my devices have finally connected. So I will see how we go today**
- **Update: Didnt take long for 7 devices to become unavailable. Repairing again**
- **Update: just had my 2nd Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms), moving to the other firmware now**

### 20230902

- **Update: Have updated my coordinator to 20230902. Did not start straight away. got this error  - 2023-09-21T02:21:23.648Z zigbee-herdsman:controller:device:log Request Queue (0xa4c138821dc3a3ed): default expiration timeout set to 0**
- Update had my first (Error: SRSP - SYS - ping after 6000ms).  Will see how we go for the rest of the day
- Update Devices still drop off the network
- Update: had my second  (Error: SRSP - SYS - ping after 6000ms). Restart and all. No im getting numerous devices just dropping off the network. Logs (error 2023-09-22 14:33:42: Exception while calling fromZigbee converter: Command 0x70ac08fffe916c19/1 manuSpecificTuya.mcuGatewayConnectionStatus({"payloadSize":1,"payload":1}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms)
- warn  2023-09-22 14:33:24: Failed to ping 'Bedroom 2 Bedside Lamp Right' (attempt 2/2, Read 0xa4c13869fe40777c/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms))

I don't think these 2 firmware are that great tbh :( im going to go back 20230716

**Comment by Koenkk** [comment]
Based on the feedback we can conclude that the 20230901/20230902 firmwares don't work well. Let's take a step back and start from 20221226, I've compiled two new test firmwares: https://github.com/Koenkk/Z-Stack-firmware/discussions/478

Please test and provide feedback, this is the only way I can provide you with a better firmware 😄
